spec/requests/memberships_controller_spec.rb

```diff
@@ -1,130 +1,282 @@
 # frozen_string_literal: true
 
 describe MembershipsController do
+  shared_context 'current_user is the group leader' do
```

Comment:
I tried reduce nesting and complexity with shared_contexts to in this test file. Any thoughts on this?

Reply:
I forgot to add to the meeting, so I do it here. I didn't know about this feature of rspec, I like the way it groups the setup for the test, but on the other hand I feel it can increase the complexity. In this actual case I think it can be avoided with the refactor.